scripts/web-crawler/index.js

```diff
@@ -539,6 +548,19 @@ function isNavigationRequest( request ) {
 	return request.isNavigationRequest() && request.frame().parentFrame() === null;
 }
 
+/**
+ * Checks, if the page is not hung by trying to evaluate a function within the page context in defined time.
+ *
+ * @param {Object} page The page instance from Puppeteer.
+ * @returns {Promise.<Boolean>}
+ */
+async function isPageResponding( page ) {
+	return Promise.race( [
+		page.title(),
+		new Promise( ( resolve, reject ) => setTimeout( () => reject(), DEFAULT_RESPONSIVENESS_CHECK_TIMEOUT ) )
```

Comment:
```suggestion
		setTimeout( () => Promise.reject(), DEFAULT_RESPONSIVENESS_CHECK_TIMEOUT )
```

or (_magic version_) 

```suggestion
		setTimeout( Promise.reject.bind( Promise ), DEFAULT_RESPONSIVENESS_CHECK_TIMEOUT )
```


will do the same.

Reply:
Yeah, I've just noticed it will not work because `Promise.race` expects an array of promises. Sorry for the confusion.